phoenix-core/src/main/java/org/apache/phoenix/coprocessor/GlobalIndexRegionScanner.java

```diff
@@ -68,30 +75,44 @@
     protected int rowCountPerTask;
     protected boolean hasMore;
     protected int maxBatchSize;
+    protected final long maxBatchSizeBytes;
+    protected final long blockingMemstoreSize;
+    protected final byte[] clientVersionBytes;
+    protected boolean useProto = true;
     protected byte[] indexMetaData;
     protected Scan scan;
     protected RegionScanner innerScanner;
     protected Region region;
     protected IndexMaintainer indexMaintainer;
     protected Table indexHTable;
     protected TaskRunner pool;
-    protected TaskBatch<Boolean> tasks;
     protected String exceptionMessage;
     protected HTableFactory hTableFactory;
     protected int indexTableTTL;
     protected long maxLookBackInMills;
-    protected IndexToolVerificationResult verificationResult;
-    protected IndexVerificationResultRepository verificationResultRepository;
+    protected IndexToolVerificationResult verificationResult = null;
+    protected IndexVerificationResultRepository verificationResultRepository = null;
+    protected Map<byte[], NavigableSet<byte[]>> familyMap;
+    protected IndexTool.IndexVerifyType verifyType = IndexTool.IndexVerifyType.NONE;
+    protected boolean verify = false;
 
     public GlobalIndexRegionScanner(RegionScanner innerScanner, final Region region, final Scan scan,
-            final RegionCoprocessorEnvironment env) throws IOException {
+                                    final RegionCoprocessorEnvironment env) throws IOException {
         super(innerScanner);
         final Configuration config = env.getConfiguration();
         if (scan.getAttribute(BaseScannerRegionObserver.INDEX_REBUILD_PAGING) != null) {
             pageSizeInRows = config.getLong(INDEX_REBUILD_PAGE_SIZE_IN_ROWS,
                     QueryServicesOptions.DEFAULT_INDEX_REBUILD_PAGE_SIZE_IN_ROWS);
         }
         maxBatchSize = config.getInt(MUTATE_BATCH_SIZE_ATTRIB, QueryServicesOptions.DEFAULT_MUTATE_BATCH_SIZE);
+        maxBatchSizeBytes = config.getLong(MUTATE_BATCH_SIZE_BYTES_ATTRIB,
+                QueryServicesOptions.DEFAULT_MUTATE_BATCH_SIZE_BYTES);
+        blockingMemstoreSize = UngroupedAggregateRegionObserver.getBlockingMemstoreSize(region, config);
+        clientVersionBytes = scan.getAttribute(BaseScannerRegionObserver.CLIENT_VERSION);
+        familyMap = scan.getFamilyMap();
+        if (familyMap.isEmpty()) {
+            familyMap = null;
```

Comment:
Any reason why we set this to null? isColumnIncluded should work if it is not null but empty right?

Reply:
It is just a little optimization for run-time checking by mapping the empty map case to the null map case so we do not check both. This check is done for every cell scanned. This code is not introduced in this PR is just moved to here.